addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/RequestorJobUnitTest.java

```diff
@@ -368,6 +368,116 @@ void shouldSendMessageWithSetValues() throws IOException {
+        job.runJob(env, progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        ArgumentCaptor<HttpMessage> argument = ArgumentCaptor.forClass(HttpMessage.class);
+        verify(httpSender).sendAndReceive(argument.capture());
+        assertThat(
+                argument.getValue().getRequestHeader().getHeader("header1"), is(equalTo("value1")));
+    }
+
+    @Test
+    void shouldSendHeaderWithEmptyStringValue() throws IOException {
+        // Given
+        AutomationProgress progress = new AutomationProgress();
+        AutomationEnvironment env = new AutomationEnvironment(progress);
+        HttpSender httpSender = mock(HttpSender.class);
+        RequestorJob job = new RequestorJob(httpSender);
+
+        String yamlStr =
+                "requests:\n"
+                        + "- url: https://www.example.com\n"
+                        + "  headers:\n"
+                        + "    header1: ''";
+        Yaml yaml = new Yaml();
+        LinkedHashMap<?, ?> jobData = (LinkedHashMap<?, ?>) yaml.load(yamlStr);
+
+        // When
+        job.setJobData(jobData);
+        job.verifyParameters(progress);
+        job.runJob(env, progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        ArgumentCaptor<HttpMessage> argument = ArgumentCaptor.forClass(HttpMessage.class);
+        verify(httpSender).sendAndReceive(argument.capture());
+        assertThat(argument.getValue().getRequestHeader().getHeader("header1"), is(equalTo("")));
+    }
+
+    @Test
+    void shouldSendHeaderWithNullValue() throws IOException {
+        // Given
+        AutomationProgress progress = new AutomationProgress();
+        AutomationEnvironment env = new AutomationEnvironment(progress);
+        HttpSender httpSender = mock(HttpSender.class);
+        RequestorJob job = new RequestorJob(httpSender);
+
+        String yamlStr =
+                "requests:\n"
+                        + "- url: https://www.example.com\n"
+                        + "  headers:\n"
+                        + "    header1: ";
+        Yaml yaml = new Yaml();
+        LinkedHashMap<?, ?> jobData = (LinkedHashMap<?, ?>) yaml.load(yamlStr);
+
+        // When
+        job.setJobData(jobData);
+        job.verifyParameters(progress);
+        job.runJob(env, progress);
+
+        // Then
+        assertThat(progress.hasErrors(), is(equalTo(false)));
+        assertThat(progress.hasWarnings(), is(equalTo(false)));
+        ArgumentCaptor<HttpMessage> argument = ArgumentCaptor.forClass(HttpMessage.class);
+        verify(httpSender).sendAndReceive(argument.capture());
+        assertThat(argument.getValue().getRequestHeader().getHeader("header1"), is(equalTo(null)));
+    }
+
+    @Test
+    void shouldNotFailIfBadHeadersList() throws IOException {
```

Comment:
`shouldAcceptEmptyHeadersList` would be more clear, this is not bad/incorrect input.

Reply:
Done